### PR TITLE
fix: make Qwen3Rerank compatible with ModuleReranker

### PR DESCRIPTION
**Pull request by ChenJiahaoST**
## Summary
- make `Qwen3Rerank` compatible with the `ModuleReranker` calling convention
- preserve the legacy `nodes + query` path while supporting direct document-list rerank calls
- filter local-only kwargs from rerank payloads and add regression tests for empty-query and empty-list edge cases

## Testing
- `conda run -n lazyllm pytest -q /Users/chenjiahao/Desktop/codes/LazyRAG-rerank-compat/tests/algorithm/test_qwen3_rerank_compat.py`

### CODE REVIEW

File: algorithm/chat/components/tmp/local_models.py [hunk 1]
```diff
@@ -153,42 +192,54 @@ def _parse_response(self, response: Any) -> List[float]:
...
+        all_scores = self._score_texts(query, texts, **kwargs)
+        scored_indices = [(index, all_scores[index]) for index in range(len(texts))]
+        scored_indices.sort(key=lambda item: item[1], reverse=True)
+        results = scored_indices[:top_k] if top_k > 0 else scored_indices
+        LOG.debug(f'Rerank use `{self._embed_model_name}` and get indices: {results}')
+        return results
+
+    def forward(self, *args: Any, **kwargs: Any) -> Union[List[DocNode], List[Tuple[int, float]]]:
+        if not args:
+            raise TypeError('Qwen3Rerank.forward() missing required positional argument')
```
[the remark this answers is not in the document]
Comment (ChenJiahaoST) [reply algorithm/chat/components/tmp/local_models.py:256]: Fixed in `eb3534b`. `forward()` now supports keyword-only callers again by handling `nodes=` and `documents=` when `args` is empty, which restores backward compatibility for calls like `reranker(query='... ', documents=[...])`. I added a regression test for the keyword-only documents path.

File: algorithm/common/model/reranker.py [hunk 2]
```diff
@@ -18,6 +18,7 @@ class Qwen3Rerank(LazyLLMOnlineRerankModuleBase):
 
     _QUERY_TEMPLATE = '{prefix}<Instruct>: {instruction}\n<Query>: {query}\n'
     _DOCUMENT_TEMPLATE = '<Document>: {doc}{suffix}'
+    _LOCAL_ONLY_PAYLOAD_KEYS = frozenset(('query', 'documents', 'template', 'top_n', 'top_k', 'topk'))
```
[the remark this answers is not in the document]
Comment (ChenJiahaoST) [reply algorithm/common/model/reranker.py]: Addressed in `eb3534b`. I expanded `_LOCAL_ONLY_PAYLOAD_KEYS` to include `nodes`, `timeout`, and `request_timeout`, so local-only kwargs no longer leak into the remote rerank payload. I also kept the existing filter for `template`/`top_n`/`topk`. Added regression coverage in `tests/algorithm/test_qwen3_rerank_compat.py`.

File: algorithm/common/model/reranker.py [hunk 3]
```diff
@@ -153,42 +192,54 @@ def _parse_response(self, response: Any) -> List[float]:
...
-                    scores += [0.0] * (len(batch_texts) - len(scores))
-                else:
-                    scores = scores[:len(batch_texts)]
-
-            all_scores.extend(scores)
-
+        all_scores = self._score_texts(query, texts, **kwargs)
         scored_nodes: List[DocNode] = [nodes[i].with_score(all_scores[i]) for i in range(len(nodes))]
         scored_nodes.sort(key=lambda n: n.relevance_score, reverse=True)
         results = scored_nodes[:top_k] if top_k > 0 else scored_nodes
```
[the remark this answers is not in the document]
Comment (ChenJiahaoST) [reply algorithm/common/model/reranker.py]: Fixed in `eb3534b`. `_rerank_nodes()` now always uses direct slicing (`scored_nodes[:top_k]`), so an explicit `top_n=0` / `topk=0` returns an empty list instead of all results. Added a regression test for the zero-topk case.

File: algorithm/common/model/reranker.py [hunk 4]
```diff
@@ -153,42 +192,54 @@ def _parse_response(self, response: Any) -> List[float]:
...
+        if not documents:
+            return []
+
+        query = self._warn_on_empty_query(query)
+        texts = [doc if isinstance(doc, str) else str(doc or '') for doc in documents]
+        top_k = self._extract_top_k(len(texts), **kwargs)
+        all_scores = self._score_texts(query, texts, **kwargs)
+        scored_indices = [(index, all_scores[index]) for index in range(len(texts))]
+        scored_indices.sort(key=lambda item: item[1], reverse=True)
+        results = scored_indices[:top_k] if top_k > 0 else scored_indices
```
[the remark this answers is not in the document]
Comment (ChenJiahaoST) [reply algorithm/common/model/reranker.py]: Fixed in `eb3534b`. `_rerank_documents()` now also uses direct slicing (`scored_indices[:top_k]`), so `top_n=0` correctly returns `[]`. Covered by the new zero-topn regression test.
